src/libraries/System.Runtime/ref/System.Runtime.cs

```diff
@@ -464,6 +479,7 @@ public abstract partial class Attribute
     {
         protected Attribute() { }
         public virtual object TypeId { get { throw null; } }
+        [System.Diagnostics.CodeAnalysis.UnconditionalSuppressMessageAttribute("ReflectionAnalysis", "IL2006:UnrecognizedReflectionPattern", Justification="Unused fields don't make a difference for equality")]
```

Comment:
Ideally we would not expose these attributes as they're useless outside of the method body. But I guess it doesn't matter that much and it will keep our tools simpler.

Reply:
If there's a concept of "list of public attributes that don't need to be propagated to ref assemblies", we should add these to the list and delete this from the ref, yes.

Reply:
There is. We should add it to:
https://github.com/dotnet/runtime/blob/master/eng/ApiCompatExcludeAttributes.txt
https://github.com/dotnet/runtime/blob/master/eng/DefaultGenApiDocIds.txt

Reply:
Awesome. Do we need it in both files? I can see it sort of fits in with the rest of the DefaultGenApi one and it seems like both files are used for the same purpose?

https://github.com/dotnet/runtime/blob/2b7852b3d039b7d40f2623a3ec54e7d23eb93790/src/libraries/Directory.Build.props#L79

Reply:
@safern can correct me if I'm wrong; I believe one is used for the tool the generates the refs and the other is used by the tool that ensures consistency between the refs and the src. I'm not sure why they're different.

Reply:
I thought the same as @stephentoub because that's what the name suggest -- it seems like they both are passed down. Maybe @ericstj adder the latter (ApiCompatExcludeAttributes.txt) for things that were there for the purpose of ApiCompat not complaining, even though it affects both... I'll defer to him on what the intention of the extra file was.